### ISSUE TITLE
`@stylistic/indentation` carries off the inline comment of a selector it re-indents

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#194]
Found in the sweep [#193](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/193) records, and set apart from it because the failure is another kind: not a fix dropped in silence but a comment destroyed.

### What minimal example or steps are needed to reproduce the bug?

```scss
a {
  b // c
.d {
    e: f;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/indentation": 2
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

The nested selector's line brought to its indentation, the comment left standing:

```scss
a {
  b // c
  .d {
    e: f;
  }
}
```

### What actually happened?

The line is brought in, and the comment is carried off:

```scss
a {
  b 
  .d {
    e: f;
  }
}
```

— what remains of `b // c` is `b ` with its trailing space, so the fix trades a comment for the very trailing whitespace other rules exist to remove, and the selector now reads `b .d`, a descendant relation the author never wrote... in fact the selector always read so to this rule, which is the point below.

### What is the cause?

The rule reads and writes `node.selector`:

```js
let selector = ruleNode.selector
…
node.selector = replaceIndentation(…)
```

and under `postcss-scss` that is the copy with every comment taken out. The other selector rules read `raws.selector.raw` and miss only the second copy ([#190](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/190), [#193](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/193)); this one reads neither, so its measurements are made against a text two characters short per comment and its write throws both raws away — PostCSS drops `raws.selector` whenever `node.selector` is assigned, and prints the assigned text.

### Do you have a proposal to fix the bug?

The reading the other rules have: take the raw where one stands, measure in it, translate reported positions through `toSelectorSourceIndex`, and write the fix to `raws.selector.raw` and its `scss` half through `restoreSelectorInlineComments`. The rule's own machinery is line-oriented rather than tree-oriented, so it wants the text path of that machinery, not the parsed one.

Styled syntax is supported in this rule alone and slated to be dropped; whatever is done here should not grow that support.
